packages/server/lib/middleware/access.middleware.ts

```diff
@@ -392,6 +392,27 @@ export class AccessMiddleware {
 
         next();
     }
+
+    internal(req: Request, res: Response, next: NextFunction) {
+        const key = process.env['NANGO_INTERNAL_API_KEY'];
+
+        if (!key) {
+            return errorManager.errRes(res, 'internal_private_key_configuration');
+        }
+
+        const authorizationHeader = req.get('authorization');
+
+        if (!authorizationHeader) {
+            return errorManager.errRes(res, 'missing_auth_header');
+        }
+
+        const receivedKey = authorizationHeader.split('Bearer ').pop();
+        if (receivedKey !== key) {
```

Comment:
if we are paranoid `crypto.timingSafeEqual` (we haven't done it for the other methods but just thought about it)